### PR TITLE
Fast integration tests refactoring

### PR DESCRIPTION
**Pull request by pbochynski**
**Description**

Changes proposed in this pull request:
- more checks for Service Catalog objects
- preparation for upgrade tests

### CODE REVIEW

**Review by aerfio — CHANGES_REQUESTED** [review]

File: tests/fast-integration/test/getting-started-guides.js [hunk 1]
```diff
@@ -102,49 +114,50 @@ describe("Getting Started Guides tests", function () {
...
         k8sDynamicApi.create(obj)
       )
     ).catch(expectNoK8sErr);
+  });
 
-    console.log("Waiting for apirule to be ready...");
-    await waitForApiRuleReady();
+  it("APIRule should be ready", async function () {
+    const apiRulePath = `/apis/gateway.kyma-project.io/v1alpha1/namespaces/${orderService}/apirules`
+    waitForK8sObject(watch, apiRulePath, {}, (_type, _apiObj, watchObj) => {
```
[the remark this answers is not in the document]
Comment (pbochynski) [reply tests/fast-integration/test/getting-started-guides.js]: fixed

File: tests/fast-integration/test/commerce-mock.js [hunk 2]
```diff
@@ -141,29 +168,37 @@ describe("Commerce Mock tests", function () {
...
+    await waitForServiceInstance(watch, 'commerce-events');
+  });
+
+  it("service binding should be ready", async function () {
+    k8sApply(k8sDynamicApi,[serviceBinding]);
+    await waitForServiceBinding(watch, 'commerce-binding');
   });
 
   it("service binding usage for function should be ready", async function () {
+    k8sDynamicApi.create(sbu);
```
[the remark this answers is not in the document]
Comment (pbochynski) [reply tests/fast-integration/test/commerce-mock.js]: fixed

File: tests/fast-integration/test/commerce-mock.js [hunk 3]
```diff
@@ -59,24 +95,20 @@ describe("Commerce Mock tests", function () {
     });
   });
 
-  it("mocks namespace should be created or fail with reason AlreadyExists ", async function () {
-    await k8sDynamicApi.create(mocksNamespaceObj).catch((e) => {
-      expect(e.body.reason).to.be.equal(alreadyExists)
-    });
+  it("mocks namespace should be created", async function () {
+    k8sApply(k8sDynamicApi, [mocksNamespaceObj]);
```
[the remark this answers is not in the document]
Comment (pbochynski) [reply tests/fast-integration/test/commerce-mock.js]: fixed

**Review by crabtree — COMMENTED** [review]
Please see my comments

**Review by crabtree — APPROVED** [review]

**Review by aerfio — APPROVED** [review]

**Pull request merged**